pkg/kubelet/rkt/rkt.go

```diff
@@ -1510,27 +1457,236 @@ func (r *Runtime) ListImages() ([]kubecontainer.Image, error) {
+					{
+						Key:   k8sRktKubeletAnno,
+						Value: k8sRktKubeletAnnoValue,
+					},
+					{
+						Key:   k8sRktUIDAnno,
+						Value: string(uid),
+					},
+				},
+			},
+		},
+	}
+
+	listResp, err := r.apisvc.ListPods(context.Background(), listReq)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't list pods: %v", err)
+	}
+
+	var latestPod *rktapi.Pod
+	var latestRestartCount int = -1
+
+	// Only use the latest pod to get the information, (which has the
+	// largest restart count).
+	for _, pod := range listResp.Pods {
+		manifest, creationTime, restartCount, err := retrievePodInfo(pod)
+		if err != nil {
+			glog.Warning("rkt: Couldn't get necessary info from the rkt pod, (uuid %q): %v", pod.Id, err)
+			continue
+		}
+
+		if restartCount > latestRestartCount {
+			latestPod = pod
+			latestRestartCount = restartCount
+		}
+
+		for i, app := range pod.Apps {
+			// The order of the apps is determined by the rkt pod manifest.
+			// TODO(yifan): Let the server to unmarshal the annotations?
+			hashStr, ok := manifest.Apps[i].Annotations.Get(k8sRktContainerHashAnno)
+			if !ok {
+				glog.Warningf("rkt: No container hash in pod manifest for rkt pod, (uuid %q, app %q)", pod.Id, app.Name)
+				continue
+			}
+
+			hashNum, err := strconv.ParseUint(hashStr, 10, 64)
+			if err != nil {
+				glog.Warningf("rkt: Invalid hash value %q for rkt pod, (uuid %q, app %q)", pod.Id, app.Name)
+				continue
+			}
+
+			cs := kubecontainer.ContainerStatus{
+				ID:    buildContainerID(&containerID{uuid: pod.Id, appName: app.Name}),
+				Name:  app.Name,
+				State: appStateToContainerState(app.State),
+				// TODO(yifan): Use the creation/start/finished timestamp when it's implemented.
+				CreatedAt: creationTime,
+				StartedAt: creationTime,
+				ExitCode:  int(app.ExitCode),
+				Image:     app.Image.Name,
+				ImageID:   "rkt://" + app.Image.Id, // TODO(yifan): Add the prefix only in api.PodStatus.
+				Hash:      hashNum,
+				// TODO(yifan): Note that now all apps share the same restart count, this might
+				// change once apps don't share the same lifecycle.
+				// See https://github.com/appc/spec/pull/547.
+				RestartCount: restartCount,
+				// TODO(yifan): Add reason and message field.
+			}
+
+			podStatus.ContainerStatuses = append(podStatus.ContainerStatuses, &cs)
+		}
+	}
+
+	if latestPod != nil {
+		// Try to fill the IP info.
+		for _, n := range latestPod.Networks {
+			if n.Name == defaultNetworkName {
+				podStatus.IP = n.Ipv4
+			}
+		}
+	}
+
+	return podStatus, nil
 }
 
-func (r *Runtime) ConvertPodStatusToAPIPodStatus(_ *api.Pod, _ *kubecontainer.PodStatus) (*api.PodStatus, error) {
-	return nil, fmt.Errorf("Not implemented yet")
+type sortByRestartCount []*kubecontainer.ContainerStatus
+
+func (s sortByRestartCount) Len() int           { return len(s) }
+func (s sortByRestartCount) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s sortByRestartCount) Less(i, j int) bool { return s[i].RestartCount > s[j].RestartCount }
```

Comment:
Hm, this is confusing - maybe we should tweak the naming of this, or have it sort actually less-than, then use sort.Reverse instead